### PR TITLE
Add details on fsspec data plugin

### PR DESCRIPTION
**Pull request by SmritiSatyanV**
Explain about fsspec and how it is helpful to the users.
Issue: https://github.com/flyteorg/flyte/issues/2049
Signed-off-by: SmritiSatyanV <smriti@union.ai>

### CODE REVIEW

File: docs/source/data.extend.rst [hunk 1]
```diff
@@ -16,3 +16,22 @@ Moreover, it is designed to be extensible and users can bring their own data per
...
+--------------------------
+
+Flytekit ships with a default storage driver that uses aws-cli on AWS and gsutil on GCP. By default, Flyte uploads the task outputs to S3 or GCS using these storage drivers.
+
+Why ``fsspec``?
+^^^^^^^^^^^^^^^
+
+You can use the fsspec plugin implementation to utilize all its available plugins with flytekit. The `fsspec <https://github.com/flyteorg/flytekit/blob/aad4e0e67fbc1614b4aa3e49c49cd777738cd3d7/plugins/flytekit-data-fsspec/setup.py#L25-L26>`_ plugin provides an implementation of the data persistence layer in Flytekit. For example: HDFS, FTP are supported in fsspec, so you can use them with flytekit too.
+The data persistence layer helps store logs of metadata and raw user data.
+As a consequence of the implementation, an S3 driver can be installed using ``pip install s3transfer``.
```
Comment (pingsutw) [inline docs/source/data.extend.rst:30]: I think it should be `s3fs`, https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=s3fs#other-known-implementations
```suggestion
As a consequence of the implementation, an S3 driver can be installed using ``pip install s3fs``.
```

Comment (SmritiSatyanV) [reply docs/source/data.extend.rst:30]: Sure, will fix that.

File: docs/source/data.extend.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ##############################
 Extend Data Persistence layer
 ##############################
-Flytekit provides a data persistence layer, which is used for recording metadata that is shared with backend Flyte. This persistence layer is available for various types to store raw user data and is designed to be cross-cloud compatible.
+Flytekit provides a data persistence layer, which is used for recording metadata that is shared with Flyte backend. This persistence layer is available for various types to store raw user data and is designed to be cross-cloud compatible.
```
Comment (marc-union) [inline docs/source/data.extend.rst:4]: Typo: "with the Flyte backend"

File: docs/source/data.extend.rst [hunk 3]
```diff
@@ -16,3 +16,22 @@ Moreover, it is designed to be extensible and users can bring their own data per
...
+
+The ``fsspec`` Data Plugin
+--------------------------
+
+Flytekit ships with a default storage driver that uses aws-cli on AWS and gsutil on GCP. By default, Flyte uploads the task outputs to S3 or GCS using these storage drivers.
+
+Why ``fsspec``?
+^^^^^^^^^^^^^^^
+
+You can use the fsspec plugin implementation to utilize all its available plugins with flytekit. The `fsspec <https://github.com/flyteorg/flytekit/blob/aad4e0e67fbc1614b4aa3e49c49cd777738cd3d7/plugins/flytekit-data-fsspec/setup.py#L25-L26>`_ plugin provides an implementation of the data persistence layer in Flytekit. For example: HDFS, FTP are supported in fsspec, so you can use them with flytekit too.
```
[the remark this answers is not in the document]
Comment (SmritiSatyanV) [reply docs/source/data.extend.rst:28]: You are right, it should be the directory, instead of the setup file.

**Review by pingsutw — APPROVED** [review]

**Pull request merged**